hooks/checkout

```diff
@@ -8,8 +8,25 @@ if [[ "$BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_SKIP_CHECKOUT" == "true" ]]; then
     exit 0
 fi
 
-WORKSPACE="$HOME/buildkite-checkouts/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID"
 echo "--- :open_file_folder: Setting up workspace"
+WORKSPACE="$HOME/buildkite-checkouts/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID"
 echo "Creating directory: $WORKSPACE"
 mkdir -p "$WORKSPACE" && cd "$WORKSPACE"
 echo "Setup completed successfully."
+
+echo "--- Checking out repository"
+echo "Setting up git and ssh..."
+# setup git to use ssh instead of https
+git config --global url."ssh://git@github.com".insteadOf "https://github.com"
```

Comment:
This would restrict user to setup an SSH key always even in case of trying to clone a public repo via https.

So, I guess we should probably directly use the CLONE_URL as such without this conversion

Reply:
then private repos will only be checked out if the `CLONE_URL` is an `ssh://` url. Do we want that constraint?

Reply:
Tested this and it seems to be problematic.  Also note that, since it is `--global` config, the setting always exists even after the build. I tried upgrading the plugin to a new version after the global flag got set and it was struggling to download the new plugin code.

(so removing this line) I am ok with the tradeoff.